### ISSUE TITLE
Charter occurs multiple times in fond overview

### ISSUE THREAD

**Issue by StephanMa** [icaruseu/mom-ca#679]
As mentioned in the email communication, the charter `A_III_3` (http://monasterium.net/mom/DE-FWUBHGW/LSU/A_III_3/charter) occurs multiple times when browsing through the fond (http://monasterium.net/mom/DE-FWUBHGW/LSU/fond).

While entering the preview of the charter and navigating in the preview, the charters just appear once.

I tested several states of the `charter-preview.widget.xml` and it seems that the error was introduced during the additions of IllUrk. I resetted the state of the widget to https://github.com/icaruseu/mom-ca/commit/395a111d753c6d1584ac80b551872a5e3cc28718#diff-6eee8c5774b738e094c2c249fe1ed314 and it works.

**Comment by GVogeler** [comment]
https://github.com/icaruseu/mom-ca/blob/master/my/XRX/src/mom/app/fond/widget/fond.widget.xml#L415 iterates through `for $year at $num in $wcharter:date-elements[ position() = $charters:start to $charters:stop ]`
which is based on `charters:years($wcharters:metadata-charter-collection)` defined in https://github.com/icaruseu/mom-ca/blob/master/my/XRX/src/mom/app/charters/charters.xqm#L78 
This function
```
let $charter-base-collection := collection('/db/mom-data/metadata.charter.public/DE-FWUBHGW/LSU/')

for $year in $charter-base-collection//cei:text//cei:issued/(cei:date/@value|cei:dateRange/@from)
    order by xs:integer($year)
    return
$year
```
 returns
`exerr:ERROR err:XQDY0025: element has more than one attribute 'from' [at line 4, column 1]`
in exide.

**Comment by GVogeler** [comment]
As the XML extracted via
```
xquery version "3.0";
declare namespace cei = "http://www.monasterium.net/NS/cei";
declare namespace atom = "http://www.w3.org/2005/Atom" ;
<result>
{

for $year in collection('/db/mom-data/metadata.charter.public/DE-FWUBHGW/LSU/')//atom:entry
    
    return
<charter id="{$year/atom:id}">{$year//cei:issued}</charter>
}
</result>
```
does not return any result to the XPath `//charter[count(.//*:dateRange/@from) gt 1]` I wonder if there might be an error in the eXist-db index.

**Comment by StephanMa** [comment]
Fond was reindexed... but error still occurs....

**Comment by StephanMa** [comment]
The problem is, that this charters contains 
´cei:date` and `cei:dateRaange` with filled in Data. The Script sees that both are filled and count this charter twice.

Proof:
deleted the entry `<cei:dateRange from="12690925" to="12690925"/>` from `tag:www.monasterium.net,2011:/charter/DE-FWUBHGW/LSU/A_III_3` and it is available just once.

**Comment by GVogeler** [comment]
Then we should consider to change https://github.com/icaruseu/mom-ca/blob/d133b9b66e96ca3c8d99d010b0593a3c41f74db7/my/XRX/src/mom/app/charters/charters.xqm#L80 in a way, that is not affected by double encoding, or make the schema stricter. A stricter schema will probably cause problems with EditMOM3 until we have solved #191 . 
I suggest to add to the XPath in L80 in charters.xqm an index `[1]` :
`for $year in $charter-base-collection//cei:text//cei:issued/(cei:date/@value|cei:dateRange/@from)[1]`

**Comment by GVogeler** [comment]
Attention: using only the index `[1]` can create wrong results:
Consider following XML:
```
<cei:issued>
  <cei:date value="99999999"/>
  <cei:dateRange from="12950101" to="12953112">Im Jahr 1295</cei:dateRange>
</cei:issued>
```
The XPath above returns `99999999`. Thus we need a a constraint like `[. lt cei:dateRange or not(../cei:dateRange)]`:

```
for $year in $charter-base-collection//cei:text//cei:issued/(
  cei:date/@value[
    not(../../cei:dateRange) 
    or . lt ../../cei:dateRange/@from
    or . = ../../cei:dateRange/@from]
  |
  cei:dateRange/@from[not(../../cei:date) or . lt ../../cei:date/@value]
 )
 ```
which will return in case of two values prefer the lower one or date/@value if both are equal.

**Comment by GVogeler** [comment]
Alternative approach would be to display the first element which contains text:
```
for $year in $charter-base-collection//cei:text//cei:issued/(cei:date/@value|cei:dateRange/@from)[normalize-space(.//text())!=''][1]
```

**Comment by GVogeler** [comment]
I've implemented the first solution with PR #701

**Comment by GVogeler** [comment]
no example for the bug exist anymore
